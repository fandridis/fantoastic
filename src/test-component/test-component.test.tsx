import React from "react";
import { shallow } from "enzyme";

import TestComponent from "./test-component";

describe("Test Component", () => {
  let props: any;

  beforeEach(() => {
    props = {
      theme: "primary"
    };
  });

  const renderWrapper = () => shallow(<TestComponent {...props} />);

  describe("Snapshots", () => {
    it("should match snapshots as primary themed", () => {
      expect(renderWrapper()).toMatchSnapshot();
    });

    it("should match snapshots as secondary themed", () => {
      props.theme = "secondary";
      expect(renderWrapper()).toMatchSnapshot();
    });
  });
});
