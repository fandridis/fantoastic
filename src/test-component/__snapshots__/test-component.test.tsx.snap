// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component Snapshots should match snapshots as primary themed 1`] = `
<div
  className="test-component test-component-primary"
>
  <h1
    className="heading"
  >
    I'm the test component
  </h1>
  <h2>
    Made with love by Harvey
  </h2>
</div>
`;

exports[`Test Component Snapshots should match snapshots as secondary themed 1`] = `
<div
  className="test-component test-component-secondary"
>
  <h1
    className="heading"
  >
    I'm the test component
  </h1>
  <h2>
    Made with love by Harvey
  </h2>
</div>
`;
